### PR TITLE
Moved 'geofences' filter to 'Filters' module and wrote unit tests for it

### PR DESCRIPTION
**Pull request by Pontiky**
## Description

I moved the `geofences` filter to `Filters` modules and wrote unit tests for it.
A lot of files changed but I needed to copy/paste the same lines of codes for each of the 8 filters types (8 filters files + 8 unit tests files).

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (would cause existing functionality to change)

## Motivation and Context

[As marked by the TODO](https://github.com/pokealarm/pokealarm/blob/90cee601de85e1cf25f97f8813ec0be07b52284d/PokeAlarm/Manager.py#L1150), I moved `geofences` filter to its right location.
This will bring a base to rewrite the `exclude_geofences` filter started by PR #738.

## How Has This Been Tested?

UNIT TESTS!!! and some others random tests with the `webhook_tester` tool.

## Wiki Update

- [ ] This change requires an update to the Wiki.
- [x] This change does not require an update to the Wiki.

### CODE REVIEW

**Review by LegitDongo — APPROVED** [review]
Everything looks good, will wait for a confirmed test before merge

**Review by billyjbryant — CHANGES_REQUESTED** [review]
@pontiky please rebase this.

**Comment by Pontiky** [comment]
> @Pontiky please rebase this.

Done

**Review by billyjbryant — APPROVED** [review]

**Pull request merged**
